Windows - Lilith 8.1.1 failed to execute from Gradle scripts

[test.zip](https://github.com/huxi/lilith/files/1717101/test.zip)
Please see the attached file for more information.

On Windows, when launching lilith.bat from Gradle script, both JAVA_HOME and JAVA_EXE are set, as a result, lilith.bat could not execute correctly and displayed below error

<pre>
C:\test > gradle lilith
> Task :lilith FAILED

WARN: JAVA_HOME is set to an invalid directory.
JAVA_HOME = C:\programs\jdk1.8.0_51
Please set the JAVA_HOME variable in your environment to match the
location of your Java installation



FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lilith'.
> Process 'command 'cmd'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
1 actionable task: 1 executed

</pre>

--- a/lilith/src/bin/lilith.bat
+++ b/lilith/src/bin/lilith.bat
@@ -38,7 +38,7 @@
 
 @REM set %HOME% to equivalent of $HOME
 if "%HOME%" == "" (set HOME=%HOMEDRIVE%%HOMEPATH%)
-if "%JAVA_EXE%" == "" (set JAVA_EXE=javaw.exe)
+
 
 set ERROR_CODE=0
 
@@ -49,19 +49,34 @@ if "%OS%"=="WINNT" @setlocal
 @REM ==== START VALIDATION ====
 if not "%JAVA_HOME%" == "" goto OkJHome
 
+@REM try JAVA_EXE 
+set LILITH_JAVA_EXE=%JAVA_EXE%
+if not "%LILITH_JAVA_EXE%" == "" if exist "%LILITH_JAVA_EXE%" goto chkLHome
+
+@REM find javaw from PATH
+set LILITH_JAVA_EXE=javaw.exe
+%LILITH_JAVA_EXE% -version >NUL 2>&1
+if "%ERRORLEVEL%" == "0" goto chkLHome
+
+
+
 echo.
-echo WARN: JAVA_HOME not found in your environment.
+echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
 echo Please set the JAVA_HOME variable in your environment to match the
 echo location of your Java installation
 echo.
-goto chkLHome
-@REM goto error
+goto error
+
+
+
+
 
 :OkJHome
-if exist "%JAVA_HOME%\bin\%JAVA_EXE%" goto chkLHome
+set LILITH_JAVA_EXE=%JAVA_HOME%\bin\javaw.exe
+if exist "%LILITH_JAVA_EXE%" goto chkLHome
 
 echo.
-echo WARN: JAVA_HOME is set to an invalid directory.
+echo ERROR: JAVA_HOME is set to an invalid directory.
 echo JAVA_HOME = %JAVA_HOME%
 echo Please set the JAVA_HOME variable in your environment to match the
 echo location of your Java installation
@@ -136,8 +151,6 @@ goto Win9xApp
 
 @REM Reaching here means variables are defined and arguments have been captured
 :endInit
-SET LILITH_JAVA_EXE="%JAVA_HOME%\bin\%JAVA_EXE%"
-if "%JAVA_HOME%" == "" (set LILITH_JAVA_EXE="%JAVA_EXE%")
 
 :runm2
 %LILITH_JAVA_EXE% %LILITH_OPTS% "-Dlilith.home=%LILITH_HOME%" -jar "%LILITH_HOME%\lib\lilith.jar" %LILITH_CMD_LINE_ARGS%

